Automattic/wp-calypso
# PrivacyPolicyBanner: remove the component and associated feature flag

**Update:** We decided to remove the `PrivacyPolicyBanner` completely, as it hasn't been used in the last 4 years and is unlikely to be used again.

This PR rewrites the `PrivacyPolicyBanner` component so that it uses React Query to do its `/privacy-policy` REST request, instead of using the legacy `http-data` framework. It also makes it a functional component, meaning that the PR is complete rewrite 🙂 

After the rewrite, I finally hide it completely behind a disabled `privacy-policy` flag. Last time the component was active was 4 years ago, in January 2018, when we had a privacy policy update that required getting a new consent from all users.

Since then, every load of Calypso Stats page downloads 100kB of data from the `/privacy-policy` endpoint, where the response contains two complete copies of the policy text, in HTML and plain text versions, only to discover it doesn't need to do anything about it.

If anyone ever changes the privacy policy version or the notification period, they'll need to toggle the flag back to `true`.

**How to test:**
Go to your site's stats page with some feature flags manually enabled:
```
http://calypso.localhost:3000/stats/day/example.blog?flags=privacy-policy,privacy-policy/test
```
and verify that you see a privacy policy banner:

<img width="618" alt="Screenshot 2022-01-06 at 14 56 29" src="https://user-images.githubusercontent.com/664258/148394438-cd0594ff-d091-4fd7-8d06-58e9e380372f.png">

and that clicking on "Learn More" opens a modal dialog with the full policy text.
